src/test/shared/codelens/localLambdaRunner.test.ts

```diff
@@ -378,4 +385,51 @@ describe('localLambdaRunner', async () => {
             })
         }
     })
+
+    describe('makeBuildDir', () => {
+        it ('creates a temp directory', async () => {
+            const dir = await localLambdaRunner.makeBuildDir()
+            assert.strictEqual(fsUtils.fileExists(dir), true)
+        })
+    })
+
+    describe('executeSamBuild', () => {
+        const failedChildProcess: ChildProcessResult = {
+            exitCode: 1,
+            error: new Error('you are already dead'),
+            stdout: 'friendly failure message',
+            stderr: 'big ugly failure message'
+        }
+
+        const successfulChildProcess: ChildProcessResult = {
+            exitCode: 0,
+            error: undefined,
+            stdout: 'everything sunny all the time always',
+            stderr: 'nothing to report'
+        }
+
+        const generateSamBuildParams = (isSuccessfulBuild: boolean) => {
+            return {
+                baseBuildDir: tempDir,
+                codeDir: tempDir,
+                inputTemplatePath: tempDir,
+                channelLogger: new FakeChannelLogger(),
+                // not needed for testing
+                manifestPath: undefined,
+                samProcessInvoker: {
+                    invoke: async (): Promise<ChildProcessResult> =>
+                        isSuccessfulBuild ? successfulChildProcess : failedChildProcess
+                }
+            }
+        }
+
+        it ('fails when the child process returns a nonzero exit code', async () => {
+            await assertRejects(async () => localLambdaRunner.executeSamBuild(generateSamBuildParams(false)))
+        })
+
+        it ('succeeds when the child process returns with an exit code of 0', async () => {
+            const samBuildResult = await localLambdaRunner.executeSamBuild(generateSamBuildParams(true))
+            assert.strictEqual(samBuildResult === path.join(tempDir, 'output', 'template.yaml'),  true)
```

Comment:
Nit: Simpler to directly assert equality on the path, and also provides a better message when the assert fails.

```typescript
assert.strictEqual(samBuildResult, path.join(tempDir, 'output', 'template.yaml'))
```

Reply:
✔️